sql/mysql/tidb.go

```diff
@@ -231,13 +262,40 @@ var reColl = regexp.MustCompile(`(?i)CHARSET\s*=\s*(\w+)\s*COLLATE\s*=\s*(\w+)`)
 func (i *tinspect) setCollate(t *schema.Table) error {
 	var c CreateStmt
 	if !sqlx.Has(t.Attrs, &c) {
-		return fmt.Errorf("missing CREATE TABLE statment in attribuets for %q", t.Name)
+		return fmt.Errorf("missing CREATE TABLE statement in attributes for %q", t.Name)
 	}
 	matches := reColl.FindStringSubmatch(c.S)
 	if len(matches) != 3 {
-		return fmt.Errorf("missing COLLATE and/or CHARSET information on CREATE TABLE statment for %q", t.Name)
+		return fmt.Errorf("missing COLLATE and/or CHARSET information on CREATE TABLE statement for %q", t.Name)
 	}
 	t.SetCharset(matches[1])
 	t.SetCollation(matches[2])
 	return nil
 }
+
+// e.g AUTO_INCREMENT=1234
+var reAI = regexp.MustCompile(`(?i)\s*AUTO_INCREMENT\s*=\s*(\d+)\s*`)
+
+// setCollate extracts the updated Collation from CREATE TABLE statement.
+func (i *tinspect) setAutoIncrement(t *schema.Table) error {
+	// patch only it is set (set falsely to '1' due to this bug:https://github.com/pingcap/tidb/issues/24702).
+	ai := &AutoIncrement{}
+	if !sqlx.Has(t.Attrs, ai) {
+		return nil
+	}
+	var c CreateStmt
+	if !sqlx.Has(t.Attrs, &c) {
+		return fmt.Errorf("missing CREATE TABLE statement in attributes for %q", t.Name)
+	}
+	matches := reAI.FindStringSubmatch(c.S)
+	if len(matches) != 2 {
+		return nil
+	}
+	v, err := strconv.ParseInt(matches[1], 10, 64)
+	if err != nil {
+		return err
+	}
+	ai.V = v
+	schema.ReplaceOrAppend(&t.Attrs, ai)
+	return nil
+}
```

Comment:
Please see https://github.com/ariga/atlas/blob/master/sql/mysql/inspect.go#L483-L508

Reply:
I saw that but I didn't understand why it needs `putShow` which isn't used by me